app/facades/food_facade.rb

```diff
@@ -0,0 +1,14 @@
+class FoodFacade
+  class << self
+    def foods_containing(keyword)
+      # binding.pry
+      FoodService.foods_containing(keyword)[:foods].each do |food_info|
+        Food.new(food_info)
+      end
```

Comment:
I think you may have gotten "each'd" 😬 , which is why you may have thought it wasn't working. If you swapped `each` out for a better case enumerable, I think this code should work.

Reply:
I got it to work! i made a mistake in my service file!

Reply:
My original response was missing the fdc.  I was moving very fast.
response = conn.get("fdc/v1/foods/search?query=#{keyword}")

Reply:
Going off of what I mentioned in your controller regarding avoiding calling two different methods that make the same api call: 

You could assign the food collection into a variable, let's say `food_collection` and the total foods count to a variable, let's say `food_count`, and then have your method return `[food_count, food_collection]`.. this return value would then be saved into the respective variables in your controller, to the left of the = sign.

Reply:
I definitely didn't even catch that, but glad you did! -- I think you'll still want to change it to a `map` though. In your view, you're accessing the JSON object since the JSON response is what's getting returned from your `foods_containing` method in your FoodFacade. Currently, in your view, since you're not using `map`, you're not working with food objects, but the data is still there because you have access to the entire response. Let me know if I can be more clear with what I'm saying, and please let me know if you have any follow up questions.